tests/codegen/paginator-tests/build.gradle.kts

```diff
@@ -19,82 +18,56 @@ buildscript {
 
 extra.set("skipPublish", true)
 
-val platforms = listOf("common", "jvm")
-
-platforms.forEach { platform ->
-    apply(from = rootProject.file("gradle/${platform}.gradle"))
-}
-
 val optinAnnotations = listOf("kotlin.RequiresOptIn", "aws.smithy.kotlin.runtime.util.InternalApi")
+kotlin.sourceSets.all {
+    optinAnnotations.forEach { languageSettings.optIn(it) }
+}
 
-kotlin {
-    sourceSets {
-        all {
-            val srcDir = if (name.endsWith("Main")) "src" else "test"
-            val resourcesPrefix = if (name.endsWith("Test")) "test-" else  ""
-            // the name is always the platform followed by a suffix of either "Main" or "Test" (e.g. jvmMain, commonTest, etc)
-            val platform = name.substring(0, name.length - 4)
-            kotlin.srcDir("$platform/$srcDir")
-            resources.srcDir("$platform/${resourcesPrefix}resources")
-            languageSettings.progressiveMode = true
-            optinAnnotations.forEach { languageSettings.optIn(it) }
-        }
-
-        val coroutinesVersion: String by project
-        val sdkVersion: String by project
-        commonMain {
-            dependencies {
-                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
-                implementation(project(":runtime:runtime-core"))
-            }
-        }
-        val jvmMain by getting {
-            kotlin.srcDir("${project.buildDir}/generated-src/src")
-        }
-    }
+kotlin.sourceSets.getByName("main") {
+    kotlin.srcDir("${project.buildDir}/generated-src/src")
+    kotlin.srcDir("${project.buildDir}/smithyprojections/paginator-tests/paginator-tests/kotlin-codegen")
 }
 
-tasks["smithyBuildJar"].enabled = false
+val coroutinesVersion: String by project
```

Comment:
inconsistent property getter location. You have `kotlinVersion` and `smithyVersion` inside the `dependencies` block but these outside. I'm not seeing a reason for the difference.